frontend/packages/pipelines-plugin/src/utils/pipeline-utils.ts

```diff
@@ -457,3 +458,17 @@ export const pipelinesTab = (kindObj: K8sKind) => {
       return null;
   }
 };
+
+export type ResultCell = { cells: string[] };
+
+export type ResultData = {
+  rows: ResultCell[];
+  columns: string[];
+};
+
+export const getCellsFromResults = (results: TektonResultsRun[]): ResultData => {
+  return {
+    rows: results.map((result) => ({ cells: [result.name, result.value] })),
+    columns: ['pipelines-plugin~Name', 'pipelines-plugin~Value'],
+  };
+};
```

Comment:
Did my idea not work? I thought PF accepts either strings or objects. I'll admit I didn't test it. I think there is value in the [Tuple](https://www.typescriptlang.org/docs/handbook/basic-types.html#tuple) though to help constrain headers and body to the same length of an array... otherwise I bet the Table will do its best and likely render more than 2 columns.

Maybe we keep the render aspects out of this utility like you did for the columns, at least that way we can keep this pure 🤔 

Thoughts?

Reply:
Keeping the controls of column in the Component makes more sense. Moved it.

Reply:
updated, yes. it works with the tuples. Did pass my local lint although I got some errors in my IDE